tooling/template-base/components/LearnerAccess/LoadBookmarks.tsx

```diff
@@ -0,0 +1,376 @@
+                      }}
+                    >
+                      <span>{t('yes')}</span>
+                    </button>
+                  </span>
+                )}
+              </div>
+            </span>
+          </div>
+        )}
+      </>
+    );
+  };
+
+  interface BookmarkItemsProps {
+    folderId: string;
+  }
+  const BookmarkItems = ({ folderId }: BookmarkItemsProps): JSX.Element | null => {
+    const { data: bookmarks, refetch: refetchBookmark } = useUserBookmarksByFolderQuery({
+      variables: { id: folderId }
+    });
+
+    if (!bookmarks || !bookmarks.UserBookmarksByFolder) return null;
+    return (
+      <>
+        {bookmarks.UserBookmarksByFolder.map(item => (
+          <BookmarkItem
+            key={item.id}
+            bookmark={item}
+            folderId={folderId}
+            refetchBookmark={refetchBookmark}
+          />
+        ))}
+      </>
+    );
+  };
+
+  type RequiredUserBookmarksByFolderQuery = Required<UserBookmarksByFolderQuery>;
+  interface BookmarkItemProps {
+    bookmark: RequiredUserBookmarksByFolderQuery['UserBookmarksByFolder'][0];
+    folderId: string;
+    refetchBookmark: UserBookmarksByFolderQueryResult['refetch'];
+  }
+  const BookmarkItem = ({ bookmark, folderId, refetchBookmark }: BookmarkItemProps) => {
+    const [listViewDropDown, setListViewDropDown] = useState<boolean>(false);
+    const [editNotes, setEditNotes] = useState<boolean>(false);
+    const [notes, setNotes] = useState<string>(bookmark.note || '');
+    const [updateBookmarkNote] = useUpdateBookmarkMutation();
+    const [destroyBookmark] = useDestroyBookmarkMutation();
+
+    const handleDelete = () => {
+      destroyBookmark({
+        variables: {
+          id: bookmark.id
+        }
+      }).then(() => {
+        refetchBookmark();
+      });
+    };
+
+    const handleSave = () => {
+      updateBookmarkNote({
+        variables: {
+          id: bookmark.id,
+          note: notes,
+          bookmarkFolder: folderId
+        }
+      }).then(() => {
+        refetchBookmark();
+        setEditNotes(!editNotes);
+      });
+    };
+
+    return (
+      <div
+        key={bookmark.id}
+        className={`odd:bg-slate-100 text-black-light py-3 px-4 bg-white-mid rounded  ${
+          listViewDropDown ? 'border-l-4 border-blue-700 pb-4' : ''
+        }`}
+      >
+        <div className="flex flex-col w-full">
+          <div className="flex flex-row">
+            <div className="flex flex-row basis-4/12">
+              {/* course title */}
+              <div className="flex items-center p-6 text-sm font-semibold font-primary">
+                {bookmark.course.title}
+              </div>
+            </div>
+            <div className="flex flex-row basis-8/12 justify-between">
+              {/* view button */}
+
+              <div className="flex items-center text-sm px-2 w-48">{bookmark.note}</div>
+              <a
+                href="'/learn/topic/' + bookmark.course.id"
```

Comment:
Don't believe this link would actually work as is (at least not without the `/redirect` at the end). Should take the approach seen in the `ti` repo for building out the href here:

https://github.com/thoughtindustries/ti/blob/f2c6536fe6773c1ea9ae141ada837243d4b5a8ab/assets/javascripts/lms/components/ti-dashboard-access-bookmarks.js#L24-L31

Reply:
possible solution pushed, would love feedback!